juju/juju
# Facade/state work for action abort

### Checklist


----

## Facade/state work for action abort

This PR encapsulates state and facade changes to deal with `juju cancel-action` aborting running actions.

Actions can enter an Aborting state when `juju cancel-action`, this will refire the action notifications watcher, allowing the uniter to handle the abort. The uniter will need to fetch the current action status hence the introduction of the ActionStatus facade call.

Second PR will follow to handle aborting action status.
Third PR will follow to handle aborting machine actions.

## QA steps

Create an action that sleeps for a few minutes.

juju run-action myapp/0 mysleepyaction
juju show-action-status action-1

Action should be running.

juju cancel-action action-1

Action should enter the aborting state, but not do anything.
Action should complete successfully after sleep has elapsed.

## Documentation changes

N/A

## Bug reference

N/A
